server.py

```diff
@@ -270,13 +271,24 @@ def asyncFinishThread(server):
 		app.logger.warning("Unable to get address info for %s." % (server["address"],))
 		return
 
+	addresses = set(data[4][0] for data in info)
 	if checkAddress:
-		addresses = set(data[4][0] for data in info)
 		if not server["ip"] in addresses:
 			app.logger.warning("Invalid IP %s for address %s (address valid for %s)."
 					% (server["ip"], server["address"], addresses))
 			return
 
+	try:
+		for addr in addresses:
+			reader = geolite2.reader()
+			geo = reader.get(addr)
+			server["geo_continent"] = geo["continent"]["code"]
+	except:
```

Comment:
ugly exception catch. Please catch exception properly not all exceptions

Reply:
All I can think of is to take `for addr in addresses:` out, but that's it.

This is and probably should be "optional" where if it doesn't work, we shouldn't crash the server thread. If people don't have the right DB files in place, it should still work.

Even if this is long in production, clients still should be able to handle servers in the list without geo hint, so this should therefore remain optional and not crashing the python stack, no matter the error. Note that the code no matter what will print out a log message indicating that there was a failure, so I don't feel that we should start handing exceptions with the depth that is suggested at https://github.com/maxmind/GeoIP2-python#database-reader-exceptions .

Or I just don't understand your suggestion... I do confess that Python is my #1 most hated and avoided language, I really don't like it and this is the most python code I've written in a afew months, and I'm convinced you and many others could do a better job with 3 minutes time.

Reply:
the problem i mention is the exception is too large, you should handle only geoip exception, not all exception which can be coding errors :)
handling it in the loop is desired :)